tanner/sessions/session_manager.py

```diff
@@ -66,21 +66,21 @@ def get_session_id(self, data):
 
         return hashlib.md5(sess_id_string.encode()).hexdigest()
 
-    async def delete_old_sessions(self, redis_client):
+    async def delete_old_sessions(self, redis_client, pg_client):
         id_for_deletion = [sess_id for sess_id, sess in self.sessions.items() if sess.is_expired()]
         for sess_id in id_for_deletion:
-            is_deleted = await self.delete_session(self.sessions[sess_id], redis_client)
+            is_deleted = await self.delete_session(self.sessions[sess_id], redis_client, pg_client)
```

Comment:
if we agree that we use redis to store intermediate sessions, why we need postgres client here?

Reply:
I think because there are lines in the `delete_session` function like:

```python
try:
await redis_client.set(sess.get_uuid(), sess.to_json())
await self.analyzer.analyze(sess.get_uuid(), redis_client, pg_client)
```
so I think it analyze them first and then in `save_sessions` function of `session_analyzer` we have deletion of the said key.

Reply:
My bad!

Reply:
But I think we can rewrite this part and make analyzer as a separate background task, to avoid using to DB connections in one place